### PR TITLE
Add helper method to determine if request can be upgraded to websocket.

### PR DESCRIPTION
**Pull request by tsegismont**
Vert.x Web And Vert.x HTTP Proxy have similar code. This method covers all cases and can be reused.

### CODE REVIEW

File: src/main/java/io/vertx/core/http/impl/HttpUtils.java [hunk 1]
```diff
@@ -983,4 +983,24 @@ public static boolean isValidHostAuthority(String host) {
...
+    if (req.version() != io.vertx.core.http.HttpVersion.HTTP_1_1) {
+      return false;
+    }
+    if (req.method() != io.vertx.core.http.HttpMethod.GET) {
+      return false;
+    }
+    MultiMap headers = req.headers();
+    for (String connection : headers.getAll(io.vertx.core.http.HttpHeaders.CONNECTION)) {
+      if (connection.toLowerCase().contains(io.vertx.core.http.HttpHeaders.UPGRADE)) {
+        for (String upgrade : headers.getAll(io.vertx.core.http.HttpHeaders.UPGRADE)) {
```
Comment (machine424) [inline src/main/java/io/vertx/core/http/impl/HttpUtils.java:997]: shouldn't we check for the exact `Upgrade: websocket`? it's cheaper and we can always address in future special cases.

**Review by machine424 — COMMENTED** [review]
Thanks Thomas!

File: src/test/java/io/vertx/core/http/Http1xTest.java [hunk 2]
```diff
@@ -5374,4 +5371,61 @@ public void testMissingHostHeader() throws Exception {
...
+    UnaryOperator<RequestOptions> config = ro -> ro.setMethod(HttpMethod.GET)
+      .putHeader(HttpHeaders.CONNECTION, HttpHeaders.UPGRADE)
+      .putHeader(HttpHeaders.UPGRADE, "foo");
+    doTestCanUpgradeToWebSocket(config, false);
+  }
+
+  private void doTestCanUpgradeToWebSocket(UnaryOperator<RequestOptions> config, boolean shouldSucceed) throws Exception {
+    server.requestHandler(req -> {
+      HttpServerResponse serverResponse = req.response();
+      int statusCode = HttpUtils.canUpgradeToWebSocket(req) ? 200 : 500;
```
[the remark this answers is not in the document]
Comment (tsegismont) [reply src/test/java/io/vertx/core/http/Http1xTest.java:5417]: It's just a test, we don't really care.

File: src/main/java/io/vertx/core/http/impl/HttpUtils.java (hunk 1 above)
Comment (tsegismont) [reply src/main/java/io/vertx/core/http/impl/HttpUtils.java:997]: No, because servers should accept any case according to the spec

**Pull request merged**